libs/device-core/src/commands/use-cases/app-backup/getAppStorageInfo.ts

```diff
@@ -0,0 +1,82 @@
+import Transport, { StatusCodes, TransportStatusError } from "@ledgerhq/hw-transport";
+import { LocalTracer } from "@ledgerhq/logs";
+import type { AppStorageInfo } from "../../entities/AppStorageInfo";
+import type { APDU } from "../../entities/APDU";
+
+/**
+ * Name in documentation: INS_APP_STORAGE_GET_INFO
+ * cla: 0xe0
+ * ins: 0x6a
+ * p1: 0x00
+ * p2: 0x00
+ * data: APP_NAME_LEN (1 byte) + APP_NAME (variable)
+ */
+const GET_APP_STORAGE_INFO: APDU = [0xe0, 0x6a, 0x00, 0x00, undefined];
+
+/**
+ * 0x9000: Success.
+ * 0x5123: Application not found.
+ * 0x662F: If the device is in recovery mode.
+ * 0x670A: Invalid application name length, two chars minimum.
+ */
+const RESPONSE_STATUS_SET: number[] = [
+  StatusCodes.OK,
+  StatusCodes.APP_NOT_FOUND_OR_INVALID_CONTEXT,
+  StatusCodes.CUSTOM_IMAGE_BOOTLOADER,
+  StatusCodes.INVALID_APP_NAME_LEN,
+];
+
+/**
+ * Retrieves the application storage information from the device.
+ *
+ * @param transport - The transport object used to communicate with the device.
+ * @param appName - The name of the application to retrieve the storage information for.
+ * @returns A promise that resolves to the application storage information.
+ * @throws {TransportStatusError} If the response status is invalid.
+ */
+export async function getAppStorageInfo(
+  transport: Transport,
+  appName: string,
+): Promise<AppStorageInfo> {
+  const tracer = new LocalTracer("hw", {
+    transport: transport.getTraceContext(),
+    function: "getAppStorageInfo",
+  });
+  tracer.trace("Start");
+
+  const params: Buffer = Buffer.concat([
+    Buffer.from([appName.length]),
+    Buffer.from(appName, "ascii"),
+  ]);
+  GET_APP_STORAGE_INFO[4] = params;
```

Comment:
[SHOULD] create a byte array like that instead of modifying the global ref, which should be a const:

`apdu = [... GET_APP_STORAGE_INFO.slice(4), params]`

Reply:
This is due to the `APDU` definition's incompatibility. In order to make it a `APDU` object, the 5th value must be defined (`Buffer` or `undefined`) which is not good. The `Lc` and `data` should be parameters and sometimes not be able to determine at the APDU definition level.

Reply:
Ok but still you shouldn't modify a constant

Reply:
It's not a constant, it's a container which should be configured at runtime.
The `const` keyword cannot be used to define an object(Array) constant, it means the object's reference is 'constant', but adding, updating and deleting properties in the object are permitted and always happen.
The possible solution here is to redefine the `APDU` object with only the first four values mandatory and constant by adding `as const` or `readonly` keyword. Then combine with the rest at runtime.

Reply:
Yes by constant I meant that we should probably add `"as const"` and also if it's in `FULL_CAPS` it usually means that it's a constant.